compiler/src/iree/compiler/Dialect/Stream/Transforms/SpecializeEncodings.cpp

```diff
@@ -5,6 +5,7 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 #include "iree/compiler/Dialect/Encoding/IR/EncodingTypes.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowTypes.h"
```

Comment:
```suggestion
// TODO(#19806): remove flow dependency when no longer snooping on the flow tensor types.
#include "iree/compiler/Dialect/Flow/IR/FlowTypes.h"
```

(we don't want stream knowing about flow)

Reply:
I think this dep can be removed with TypeInterface change. I have implemented it in https://github.com/iree-org/iree/pull/19700

It is needed in the PR because I need to access Flow::DispatchTensorType. In the other PR, the type implements the type interface, which allows us to query/update encoding through interface methods.

Reply:
Removed the TODO because now we can update the encodings in "preprocessing" and use them directly in the duplication.